pandas/core/arrays/base.py

```diff
@@ -777,8 +777,8 @@ def convert_values(param):
             if coerce_to_dtype:
                 try:
                     res = self._from_sequence(res)
-                except TypeError:
-                    pass
+                except Exception:
+                    res = np.asarray(res, dtype=object)
```

Comment:
Hmm, this looks like it's in the wrong place.

Right now with `coerce_to_dtype=False` we'd return an list, not an ndarray. I assume we want an ndarray of objects regardless of `coerce_to_dtype`? I don't want the return type to be `Union[ExtensionArray, List, ndarray]`

Reply:
Grr... ok what do to here?

If we do

```
                res = np.asarray(res, dtype=object)
```

we'll start failing some other tests like `Series[DecimalArray] == 1`, since we get an object dtype instead of bool. So we just leave it up to NumPy's regular inference then, `res = np.asarray(res)`?